### PR TITLE
Add unit tests for generate_gcnv_bed.py

### PR DESCRIPTION
**Pull request by jethror1**
- adds full unit tests for `generate_gcnv_bed.py`
- set appropriate dtypes in `read_single_copy_ratio_file()` for speed / lower memory usage
- anonymise other sample traces in output with `.` instead of a space
- add minimal test files
- add GitHub action to run pytest
- add `.gitginore` to hide the mess of local files

- - -
This change is [Reviewable](https://reviewable.io/reviews/eastgenomics/eggd_GATKgCNV_call/21)

## Release Notes

- **New Features**
	- Introduced a new GitHub Actions workflow for automated testing.
	- Enhanced script logging and error handling in the genomic data processing script.

- **Bug Fixes**
	- Improved error handling in the data processing script to provide clearer messages.

- **Documentation**
	- Updated the title in the README for better visibility.

- **Chores**
	- Expanded the `.gitignore` file to exclude unnecessary files generated during development.
	- Added configuration to the `.coveragerc` file for focused coverage analysis.
	- Created a suite of unit tests for the `generate_gcnv_bed` module to ensure functionality.
	- Organised the paths and logging in the genomic data processing script for clarity.

### CODE REVIEW

**Review by Yu-jinKim — APPROVED** [review]
> Reviewed 12 of 12 files at r1, all commit messages.
> *[Reviewable](https://reviewable.io/reviews/eastgenomics/eggd_GATKgCNV_call/21)* status: all files reviewed, 4 unresolved discussions (waiting on @chrispyatt and @jethror1)

**Review by chrispyatt — APPROVED** [review]
> Reviewed all commit messages.
> *[Reviewable](https://reviewable.io/reviews/eastgenomics/eggd_GATKgCNV_call/21)* status: all files reviewed, 4 unresolved discussions (waiting on @jethror1)

File: .gitignore [hunk 1]
```diff
@@ -1,3 +1,161 @@
...
+# Cython debug symbols
+cython_debug/
+
+# project specific ignores
+output/*
 *.bed.gz
 *.tbi
-*.tsv
\ No newline at end of file
+./*.tsv
```
[the remark this answers is not in the document]
Comment (jethror1) [reply .gitignore]: Done.

File: .github/workflows/pytest.yml [hunk 2]
```diff
@@ -0,0 +1,39 @@
...
+
+jobs:
+  build:
+    runs-on: ubuntu-latest
+    steps:
+    - uses: actions/checkout@v2
+    - name: Set up Python 3.8
+      uses: actions/setup-python@v1
+      with:
+        python-version: 3.8
```
[the remark this answers is not in the document]
Comment (jethror1) [reply .github/workflows/pytest.yml]: Done.

File: .github/workflows/pytest.yml [hunk 3]
```diff
@@ -0,0 +1,39 @@
...
+    - name: Install dependencies
+      run: |
+        python -m pip install --upgrade pip
+        pip install pipenv codecov
+        pip install pipenv codecov pytest pytest-cov pytest-metadata pytest-mock pytest-subtests six
+        pip install --no-index --no-deps resources/home/dnanexus/packages/*
+        pipenv install --dev
```
[the remark this answers is not in the document]
Comment (jethror1) [reply .github/workflows/pytest.yml]: Done.

File: resources/home/dnanexus/generate_gcnv_bed/tests/test_generate_gcnv_bed.py [hunk 4]
```diff
@@ -0,0 +1,736 @@
...
+from ..generate_gcnv_bed import (
+    calculate_mean_and_std_dev,
+    compress_and_index_bed_file,
+    main,
+    parse_args,
+    read_single_copy_ratio_file,
+    read_all_copy_ratio_files,
+    write_run_level_bed_file,
+    write_sample_bed_file,
+)
```
[the remark this answers is not in the document]
Comment (jethror1) [reply resources/home/dnanexus/generate_gcnv_bed/tests/test_generate_gcnv_bed.py:30]: Done.

**Review by jethror1 — COMMENTED** [review]
> *[Reviewable](https://reviewable.io/reviews/eastgenomics/eggd_GATKgCNV_call/21)* status: 11 of 12 files reviewed, 4 unresolved discussions (waiting on @coderabbitai[bot] and @Yu-jinKim)

**Pull request merged**